src/main/java/model/query/ComplexQueryDecoder.java

```diff
@@ -0,0 +1,27 @@
+package model.query;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ComplexQueryDecoder implements QueryDecoder {
```

Comment:
public class ComplexQueryDecoder implements QueryDecoder {

    private final static String INCLUDE_CHAR = "+";
    private final static String EXCLUDE_CHAR = "-";
    private final static String SEPRATOR = " ";

    @Override
    public HashMap<String,Set<String>> decode(String queryRequest) {
        String[] args = queryRequest.split(SEPRATOR);
        return new HashMap<>() {{
            put("including", decode(args, INCLUDE_CHAR));
            put("excluding", decode(args, EXCLUDE_CHAR));
            put("intersection", decode(args, Set.of(INCLUDE_CHAR, EXCLUDE_CHAR)));
        }};
    }

    @Override
    public Set<String> decode(String[] args, String starter) {
        args.stream.filter(x -> x.startsWith(starter)).toSet();
    }

    @Override
    public Set<String> decode(String[] args, Set<String> notStarter) {
        args.stream.filter(x -> !x.startsWith(starter)).toSet();
    }
}